### ISSUE TITLE
Make t.p.policies.TimeoutMixin and t.i.base.DelayedCall more unittest-friendly

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#11501]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#1722|
|Type|enhancement|
|Created|2006-05-17 20:08:54Z|

By parameterizing ```reactor.callLater``` on ```TimeoutMixin``` and ```seconds``` on ```DelayedCall```, tests of code using these or based on these can actually be deterministic.

<details><summary>Searchable metadata</summary>

```
trac-id__1722 1722
type__enhancement enhancement
reporter__exarkun exarkun
priority__highest highest
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1147896534000000 1147896534000000
changetime__1147966036000000 1147966036000000
version__ 
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set owner to @radix|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)|@radix set owner to @exarkun|
|-|-|

Is there a reason you didn't put the conditional on the seconds argument in __init__, and just assign the global seconds to self.seconds, instead of in two places in DelayedCall?

other than that, looks good.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun commented|
|-|-|

Late binding!  Probably no one depends on this, but it's difficult to be sure.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set owner to @radix|
|-|-|

Merged forward, added epytext markup for the DelayedCall.__init__ seconds parameter.  Left seconds the same, but maybe I am being overly cautious?

**Comment by twisted-trac** [comment]
|[<img alt="radix's avatar" src="https://avatars.githubusercontent.com/u/227068?s=50" width="50" height="50">](https://github.com/radix)|@radix set owner to @exarkun|
|-|-|

That's fine. I assume the use case you are expecting is "base.seconds = hork"? anyway, go ahead and merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set status to `closed`|
|-|-|

(In [16859]) Merge parameterize-some-time-1722-2

Author: exarkun
Reviewer: radix
Fixes [#11501](https://github.com/twisted/twisted/issues/11501)

This adds an optional seconds parameter to DelayedCall which, if passed, is used to
tell time in place of twisted.python.runtime.seconds.

This also adds a callLater method to TimeoutMixin which can be overridden to write
deterministic tests more easily.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
